### ISSUE TITLE
Some image link are using absolute url instead of relatively url

### ISSUE THREAD

**Issue by ericwongcm** [boxbilling/boxbilling#306]
I have originally installed boxbilling under a subdomain but I have since moved its database to another location while still using the same database.

I noticed two of the images are not showing up and when I look at the html source code, I can see the reason is that those images are still pointing to the previous subdomain which I have already deleted.

1) The spamchecker icon under settings for admin (Btw, there is a typo there too, it is reading "Spam Cheker", missing the c in Checker.)
2) The image link for cloud-host-3.png on the order page for customer, i.e. /index.php?_url=/order

I have looked everywhere inside Configuration -> Settings but I don't see anywhere I can change the url for boxbilling either, so I can get these links fixed.

Please advice how I can get this fixed.

Lastly, I am using BoxBilling 4.16.2 with boxbilling theme.

**Comment by vaidas-lungis** [comment]
Extension icon links are stored in database extension table manifest field. And yes icon link is absolute url. We will make adjustments in the next release.

Now you can edit database manually or just disable and enable extension. It should generate correct absolute url for extension icon.

**Comment by ericwongcm** [comment]
Actually, what I am mainly concerned with is the image cloud-host-3.png on the order page for customer because users will see it. May I know where I can modify this absolute url in the database?

Btw, enabling/disabling Spamchecker doesn't fix the absolute url problem.

**Comment by vaidas-lungis** [comment]
Can you elaborate more on cloud-host-3.png image ? Where do you use it and how did you inserted in the code? Please give more details to better assist you

**Comment by ericwongcm** [comment]
If you visit this page in boxbilling installation, 
http://yourdomain.com/index.php?_url=/order

You will see an image below Select product which is using the image cloud-host-3.png.

**Comment by vaidas-lungis** [comment]
Do you have created custom product?

**Comment by ericwongcm** [comment]
Yes, I have created a custom category. Since you asked, I checked and I just noticed the reason is because the icon in the custom category is using absolute url for the icon. I see I am able to change the url by editing the category.

I would suggest the custom category icon url to be changed to relatively url to avoid broken links when one migrate the boxbilling installation and database to different paths.

**Comment by vaidas-lungis** [comment]
As you can see you can use default icon from list or just paste absolute url of image in product Icon/Image URL field. You can use image from external source for that reason absolute url is used here.

**Comment by ericwongcm** [comment]
Well, seems like I can't use relative url in this url field. I will end up with invalid url in the resulting html page's source code. Maybe boxbilling should be using relative url here unless someone is specifically uses absolute url here...

**Comment by vaidas-lungis** [comment]
Yes you are right it could be relative url and a radio box to switch it for absolute url if someone requires it. But at the moment this field value must be absolute url

**Comment by 4Giedrius** [comment]
I have fixed this issue by using relative path for product's icons (https://github.com/boxbilling/boxbilling/commit/af86bd52cc992a18bedaf9f6905921175a863f27), however you will have to change existing URLs in database.

**Comment by ericwongcm** [comment]
I have just updated to boxbilling 4.18
I see a lot of broken images again.

Broken images are found on
/bb-admin/extension
/bb-admin/system

When I examine the underlying code, I noticed the reason is due to extra backslash (/).

My boxbilling is installed in a folder of the domain name, e.g. test.com/boxbilling

For those images that show up correctly, the code looks like
<img src="images/icons/middlenav/cog.png" alt="Notification center" style="width: 32px; height: 32px;">

For those that does not show up, the code looks like
<img src="/bb-modules/Massmailer/icon.png" alt="Mass mail tool" style="width: 32px; height: 32px;">

The extra backslash means it the image url becomes http://test.com/image.png instead of http://test.com/boxbilling/image.png
